dask/dataframe/backends.py

```diff
@@ -45,6 +46,148 @@
+        **kwargs :
+            Optional backend kwargs.
+
+        See Also
+        --------
+        dask.dataframe.io.io.from_dict
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def read_parquet(path: str | list, **kwargs):
+        """Read Parquet files into a DataFrame collection
+
+        Parameters
+        ----------
+        path : str or list
+            Source path(s).
+        **kwargs :
+            Optional backend kwargs.
+
+        See Also
+        --------
+        dask.dataframe.io.parquet.core.read_parquet
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def read_json(url_path: str | list, **kwargs):
+        """Read json files into a DataFrame collection
+
+        Parameters
+        ----------
+        url_path : str or list
+            Source path(s).
+        **kwargs :
+            Optional backend kwargs.
+
+        See Also
+        --------
+        dask.dataframe.io.json.read_json
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def read_orc(path: str | list, **kwargs):
+        """Read ORC files into a DataFrame collection
+
+        Parameters
+        ----------
+        path : str or list
+            Source path(s).
+        **kwargs :
+            Optional backend kwargs.
+
+        See Also
+        --------
+        dask.dataframe.io.orc.core.read_orc
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def read_csv(urlpath: str | list, **kwargs):
+        """Read CSV files into a DataFrame collection
+
+        Parameters
+        ----------
+        urlpath : str or list
+            Source path(s).
+        **kwargs :
+            Optional backend kwargs.
+
+        See Also
+        --------
+        dask.dataframe.io.csv.read_csv
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def read_hdf(pattern: str | list, key: str, **kwargs):
+        """Read HDF5 files into a DataFrame collection
+
+        Parameters
+        ----------
+        pattern : str or list
+            Source path(s).
+        key : str
+            Group identifier in the store.
+        **kwargs :
+            Optional backend kwargs.
+
+        See Also
+        --------
+        dask.dataframe.io.hdf.read_hdf
+        """
+        raise NotImplementedError
+
+
+class DataFrameCreationDispatch(CreationDispatch):
+    def register_backend(
+        self, name: str, backend: BackendEntrypointType
+    ) -> BackendEntrypointType:
+        """Register a target class for a specific dataframe-backend label"""
+        if not isinstance(backend, DataFrameBackendEntrypoint):
+            raise ValueError(
+                f"DataFrameCreationDispatch only supports "
+                f"DataFrameBackendEntrypoint registration. "
+                f"Got {type(backend)}"
+            )
+        self._lookup[name] = backend
+        return cast(BackendEntrypointType, backend)
```

Comment:
This seems 90% duplicated from `ArrayCreationDispatch` -- it may be possible to move this into the base class with a strategic class var for the `BackendEntrypointType`

Reply:
Good point - I'll try to remove the duplication.